pkg/plugins/gateway/algorithms/prefix_cache.go

```diff
@@ -49,41 +58,60 @@ func init() {
 
 type prefixCacheRouter struct {
 	cache              cache.Cache
-	tokenizer          tokenizer.Tokenizer
+	tokenizer          tokenizer.Tokenizer // Fallback tokenizer for backward compatibility
+	tokenizerPool      *TokenizerPool      // Model-aware tokenizer pool
 	prefixCacheIndexer *prefixcacheindexer.PrefixHashTable
 }
 
 func NewPrefixCacheRouter() (types.Router, error) {
-	// Create tokenizer based on type
+	c, err := cache.Get()
+	if err != nil {
+		klog.Error("fail to get cache store in prefix cache router")
+		return nil, err
+	}
+
+	// Create fallback tokenizer based on type
 	// Supported tokenizers: ["character", "tiktoken"]
 	// Default: "character" for any unrecognized type
-	// TODO: Add support for "remote" and "vllm" tokenizer types in a future PR.
-	//       This will require proper configuration handling for remote endpoints.
-	var tokenizerObj tokenizer.Tokenizer
+	var fallbackTokenizer tokenizer.Tokenizer
 	if tokenizerType == "tiktoken" {
-		tokenizerObj = tokenizer.NewTiktokenTokenizer()
+		fallbackTokenizer = tokenizer.NewTiktokenTokenizer()
 	} else {
 		// Default to character tokenizer for backward compatibility
 		if tokenizerType != "character" {
 			klog.InfoS("unrecognized tokenizer type, defaulting to character", "type", tokenizerType)
 		}
-		tokenizerObj = tokenizer.NewCharacterTokenizer()
+		fallbackTokenizer = tokenizer.NewCharacterTokenizer()
 	}
 
-	c, err := cache.Get()
-	if err != nil {
-		klog.Error("fail to get cache store in prefix cache router")
-		return nil, err
+	// Initialize TokenizerPool for vLLM remote tokenizer support
+	poolConfig := TokenizerPoolConfig{
+		EnableVLLMRemote:     enableVLLMRemoteTokenizer,
+		EndpointTemplate:     vllmTokenizerEndpointTemplate,
```

Comment:
we probably need to add model -> engine mapping from `model.aibrix.ai/engine: "vllm"` in the cache so consumer can easily distinguish engine

Reply:
/cc @happyandslow @varungup90